### PR TITLE
barebones for SequentialChain

### PR DESCRIPTION
**Pull request by ITP2023**
A starter addition for SequentialChain. It has not actually been tested against test cases.
Recommended further review and testing.

reference:
- https://github.com/hwchase17/langchain/blob/master/langchain/chains/sequential.py
- https://github.com/hwchase17/langchainjs/blob/main/langchain/src/chains/sequential_chain.ts

### CODE REVIEW

**Comment by davidmigloz** [comment]
thanks for the PR @ITP2023 !
I'll review it soon.

File: packages/langchain/lib/src/chains/sequential.dart [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      if (missingKeys.isNotEmpty) {
+        throw LangChainException(message: 'The following output variables were expected to be in the final chain output but were not found: $missingKeys');
+      }
+    }
+  }
+}
+
+
+class SimpleSequentialChain extends BaseChain {
+  // Simple chain where the outputs of one step feed directly into next.
```
Comment (davidmigloz) [inline packages/langchain/lib/src/chains/sequential.dart:94]: This comment should be a [docummentation comment](https://dart.dev/language/comments#documentation-comments).

Maybe we can also explain the difference between `SimpleSequentialChain` and `SequentialChain`.

From LangChain docs bot:
> SimpleSequentialChain is the simpler form of sequential chains, where each step in the chain has a singular input and output. It is suitable for scenarios where you want to pass the output of one step as the input to the next step in a straightforward manner.

> On the other hand, SequentialChain is a more general form of sequential chains that allows for multiple inputs and outputs. It is useful for more complex scenarios where you need to handle multiple variables and have more flexibility in the flow of data between steps.

File: packages/langchain/lib/src/chains/sequential.dart [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      }
+    }
+    return {
+      outputKey: _input
+    };
+  }
+
+  @override
+  Future<ChainValues> call(input, {bool returnOnlyOutputs = false}) {
+    return super.call(input, returnOnlyOutputs: returnOnlyOutputs);
```
Comment (davidmigloz) [inline packages/langchain/lib/src/chains/sequential.dart:145]: do we need this?

File: packages/langchain/lib/src/chains/sequential.dart [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  }
+}
```
Comment (davidmigloz) [inline packages/langchain/lib/src/chains/sequential.dart:147]: We need to add some unit tests as well.
Here you can find the tests from the original project:
https://github.com/hwchase17/langchain/blob/master/tests/unit_tests/chains/test_sequential.py

I normally port the Python test to Dart using ChatGPT with the following prompt:
````
Convert the following Python test to Dart.
Use single quotes for Strings.
Use the following format for Dart tests:
```dart
test('Name of the test', () {
  expect(value1, value2);
});
```

Python code:
```python

```
````

**Comment by davidmigloz** [comment]
hey @ITP2023 ,

Thank you so much again for your contribution!
I've added some review comments, let me know if you have any questions regarding them.

Cheers.

**Comment by ITP2023** [comment]
I've updated this branch.Please review @davidmigloz

**Comment by davidmigloz** [comment]
awesome! I'll review it tomorrow

**Comment by davidmigloz** [comment]
Congrats @ITP2023, the PR is merged 🚀

I made some fixes and improvements. Check them out to take them into account in future 🙂

Thanks again for your contribution!

**Pull request merged**